### ISSUE TITLE
many_up_to_n missing?

### ISSUE THREAD

**Issue by uucidl** [UpstandingHackers/hammer#134]
Hammer has:

| hammer   | ABNF      |
|----------|-----------|
| h_many   | *(expr)   |
| h_many1  | 1*(expr)  |
| h_repeat | n*n(expr) |
| ?        | *n(expr)  |
| ?       | i*j(expr) |

I've searched the source code and to my knowledge the last two cases are missing, for expressions with many elements `up to` `n` or many between `i` and `j`.

**Comment by abiggerhammer** [comment]
Good point! What would you think about folding the two missing cases into an `h_many_between` that takes a lower and upper bound? `up_to_n` is like saying `between(0, n)`.

**Comment by uucidl** [comment]
Yeah that would be perfect!

**Comment by uucidl** [comment]
By the way this came up as I was building an RFC5322 parser from the spec. The lines in the body are all supposed to be `*998text`

**Comment by abiggerhammer** [comment]
Cool. I'll try and get that banged into many.c this weekend.
